rpc/rpcserver.go

```diff
@@ -125,44 +139,42 @@ func (r *rpcServer) UpdateGateway(ctx context.Context, in *rpcapi.UpdateGatewayR
 		return nil, fmt.Errorf("failed to create ATX provider: %w", err)
 	}
 
-	// Close the old connections and save the new manager.
-	// The temporary manager is nil-ed to avoid closing connections in defer.
-	r.gtwManager.Close()
-	r.gtwManager, gtwManager = gtwManager, nil
+	// Swap the new and old gateway managers.
+	// The old one will be closed in defer.
```

Comment:
I can see this causing issues in the future if someone isn't aware that the `defer` statement is executed on the new OR the old `gtwManager` depending on when it is executed.

There are only two returns between `defer` and here. I would prefer calling `Close()` explicitly before these returns and here call close on the old `gtwManager`.

It's even fever lines of code (3 x `gtwManager.Close()`) instead of 5 lines with the `defer` statement. Let's not be more clever with the code than we need to be 🙂

Reply:
Manually calling `Close()` before every `return` brings a risk that in the future, somebody refactoring the code will not be aware `Close()` must be called or will forget to do it.

I don't agree it's "clever code". It's the best one can do given that Go doesn't have RAII/destructors to make sure that resources are eventually deallocated.